### PR TITLE
add CDO variant for BY strategy

### PR DESCRIPTION
**Pull request by massun-onibakuchi**

### CODE REVIEW

**Comment by massun-onibakuchi** [comment]
Somehow the setup script fails immediately after the line 74.
image
image

**Review by bugduino — COMMENTED** [review]
Left some comments. Regarding the `approve` issue I think it may be related to the fact that USDT is not a standard ERC20 token, (`approve` and `transfer` do not return `true`) so we need to use the `safeApprove` instead in the `setup` method

**Comment by bugduino** [comment]
The issue is caused because in the `testInitialize` you are calling `this.testInitialize` which causing the loop. We should change the  `testInitialize` to `public` in `TestIdleCDOBase` and the in the overriding function call `super.testInitialize` instead

File: contracts/IdleCDOBestYieldVariant.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+pragma solidity 0.8.10;
+
+import "./IdleCDO.sol";
+import "./interfaces/IERC20Detailed.sol";
+
+/// @title IdleCDO variant as proxy for Idle best-yield
+/// @author Idle DAO, @massun-onibakuchi
+contract IdleCDOBestYieldVariant is IdleCDO {
+    function _additionalInit() internal override {
+        delete isAYSActive; // disable yield split
```
[the remark this answers is not in the document]
Comment (massun-onibakuchi) [reply contracts/IdleCDOBestYieldVariant.sol:11]: right, but I find that setting `allowAAWithdraw` to `false` doesn't work when the CDO is not paused.

```
  function withdrawAA(uint256 _amount) external returns (uint256) {
    require(!paused() || allowAAWithdraw, '3');
    return _withdraw(_amount, AATranche);
  }
```

**Comment by massun-onibakuchi** [comment]
> The issue is because in `testInitialize` you are calling `this.testInitialize` which causes the loop. We should change the `testInitialize` to `public` in `TestIdleCDOBase` and then in the overriding function call `super.testInitialize` instead

oh i missed the loop. got it

File: contracts/IdleCDOBestYieldVariant.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (bugduino) [reply contracts/IdleCDOBestYieldVariant.sol]: ah ok you are right!

**Pull request merged**